updating packages

iruzevic opened the pull request:
I have refactored a lot of stuff. Before you check this PR we should discus what I did and why also I will check rename script because there were a lot of changes in code.

iruzevic commented:
I have also updated readme file with all details how to use and run boilerplate

dingo-d commented on README.md:
````diff
@@ -1,222 +1,166 @@
 
-We are using [Infinum coding standards for WordPress](https://github.com/infinum/coding-standards-wp) to check php files.
+    ```bash
+    npm run precommit
````
Can we setup `githooks` here instead of manually running scripts? [Githooks](https://githooks.com/)

iruzevic replied:
sure we can add that after we test it out. lets set this for version 3.1?

dingo-d commented on README.md:
```diff
@@ -1,222 +1,166 @@
+2. **Linting PHP** - We are using [Infinum coding standards for WordPress](https://github.com/infinum/coding-standards-wp) to check php files. To install it, you need to install [Composer](https://getcomposer.org/) from initial setup step. 
 
-* Add this aliases to you bash config:
+3. **Linting PHP** - If you have composer installed add this aliases to you bash config and reload terminal:
```
We should rephrase this to be: `you can add these aliases to your bash/zsh config for easier usage`.

iruzevic replied:
👍

dingo-d commented on README.md:
````diff
@@ -1,222 +1,166 @@
+    bash bin/build.sh
+    ```
+
+3. When rsyncing to server use `ci-exclude.txt` to exclude unnecesery folders and files on server.
````
Add a phrase: 'For deployment, we are using [rsync](https://rsync.samba.org/) in ore continuous deployment setup', and maybe add a sample example.

iruzevic commented:
packages updated, fixed creation of source maps, minor bug in class-media, and replacing normalize.css with sass version for lighter output, and fixed all terminal warnings

please check end merge this. so we can release version 3 tnx @dingo-d

iruzevic commented:
updating composer and fixing all linting errors and warnings

dingo-d commented on README.md:
````diff
@@ -1,222 +1,166 @@
+**Run this only once!**
 
-If you are using VVV clone it in the `public_html` folder
+    ```bash
````
rename should be done either with shell script or using node script. In order for node script to work, `npm install` should be ran and then `npm run rename` iirc. This should be added to the readme.

iruzevic replied:
you run sh bin/rename.sh and that runs 
````
npm install
npm run rename
````

this shell script should run on windows and osx

dingo-d replied:
I'll try this out on Windows :+1:

dingo-d commented on README.md:
````diff
@@ -1,222 +1,166 @@
+    ```bash
+    alias phpcs='vendor/bin/phpcs';
+    alias phpcbf='vendor/bin/phpcbf';
+    alias wpcs='phpcs --standard=vendor/infinum/coding-standards-wp/Infinum';
````
In the case where phpcs is run using the vendor version, the standard isn't necessary since it will pick up the `phpcs.xml.dist` as the configuration. In that file we have specified the Infinum's coding standard as a default. So the last two aliases are not really needed 😉

iruzevic replied:
ok I will remove it then.

dingo-d commented on bin/rename.js:
```diff
@@ -39,25 +39,32 @@ const findReplace = (findString, replaceString) => {
 
 // Main script
-consoleOutput(fgGreen, 'The script will uniquely set up your theme.');
+consoleOutput(fgGreen, 'Welcome to Boilerplate readme script. The script will uniquely set up your theme.');
```
> Welcome to Boilerplate readme script

You meant `Welcome to Boilerplate rename script` ? 😄

iruzevic replied:
maybe :D

dingo-d commented on wp-content/themes/inf_theme/admin/class-admin.php:
```diff
@@ -18,49 +17,28 @@
-    $main_style = $this->general_helper->get_manifest_assets_data( 'applicationAdmin.css' );
-    wp_register_style( static::THEME_NAME . '-style', $main_style );
+    $main_style = General_Helper::get_manifest_assets_data( 'applicationAdmin.css' );
+    wp_register_style( static::THEME_NAME . '-style', $main_style, array(), static::THEME_VERSION );
```
Wouldn't `static::THEME_VERSION` affect cache busting we are doing using `manifest.json`?

I know that there is an error that happens because theme version isn't specified, but there is an [upstream issue](https://github.com/WordPress-Coding-Standards/WordPress-Coding-Standards/issues/1472) opened at WPCS about this (they neglected the fact that hashing can be used).

You can either disable these lines with an inline comment 

`// phpcs:ignore WordPress.WP.EnqueuedResourceParameters.MissingVersion`

Or by adding this in the `phpcs.xml.dist` ruleset and disabling it.

iruzevic replied:
I have tested this and we are ok, assets manifest changes the name of the output file every time something changes so, everytime we hit build name of the file will change and cache busting will be done.

dingo-d replied:
Awesome, I thought that specifying version would suppress it. :+1:

dingo-d commented on wp-content/themes/inf_theme/admin/class-admin.php:
```diff
@@ -18,49 +17,28 @@
-    $main_script = $this->general_helper->get_manifest_assets_data( 'applicationAdmin.js' );
-    wp_register_script( static::THEME_NAME . '-scripts', $main_script );
+    $main_script = General_Helper::get_manifest_assets_data( 'applicationAdmin.js' );
+    wp_register_script( static::THEME_NAME . '-scripts', $main_script, array(), static::THEME_VERSION ); // phpcs:ignore WordPress.WP.EnqueuedResourceParameters.NotInFooter
```
Why aren't we loading the scripts in the footer? I thought that it's better to load them all in the footer.

iruzevic replied:
because this is up to the developer to decide when to load scripts we can load them in the footer in the boilerplate.

dingo-d commented on wp-content/themes/inf_theme/admin/class-admin.php:
```diff
@@ -71,7 +49,7 @@ public function enqueue_scripts() {
-   * @since 2.1.0
+   * @since 1.0.0
    */
   public function set_admin_color_based_on_env( $color_scheme ) {
```
Not sure we have this in the wiki, but it should be added, explaining what it does and why.

iruzevic replied:
We don't have a lot of stuff in wiki, I will update this at some point

dingo-d replied:
Or maybe @IvanGrginovInf can, since he has the gift of really writing things out in detail 😄😛 

seeWhatIDidThere xD

dingo-d replied:
It's never too late to contribute :D

dingo-d commented on wp-content/themes/inf_theme/admin/class-media.php:
```diff
@@ -51,10 +28,10 @@ public function add_theme_support() {
    */
   public function add_custom_image_sizes() {
-    add_image_size( 'full_width', 9999, 9999, false );
+    add_image_size( 'full-width', 1440, 9999, false );
```
Wouldn't this limit the width to 1440px on full-width images? What if someone wants to put 2000px image in the background or something like that?

dingo-d replied:
I agree that we should encourage optimization, but limiting this could cause issues opened with: why is my image not full size? :D Setting it to `9999` basically means, it will be whatever you put it (unless you put 30k px image xD)

iruzevic replied:
if this is an issue then I will remove it because this actually does nothing. if you upload the image it will resize it to its max size which is its uploaded size.  It is the same as full.

> the_post_thumbnail( 'full' );          // Full resolution (original size uploaded)

dingo-d commented on wp-content/themes/inf_theme/theme/class-theme.php:
```diff
@@ -17,36 +16,15 @@
-    $main_style = $this->general_helper->get_manifest_assets_data( 'application.css' );
-    wp_register_style( static::THEME_NAME . '-style', $main_style );
+    $main_style = General_Helper::get_manifest_assets_data( 'application.css' );
+    wp_register_style( static::THEME_NAME . '-style', $main_style, array(), static::THEME_VERSION );
```
Same comment as above about versions and cache busting.

iruzevic replied:
answered ⬆️

dingo-d commented on wp-content/themes/inf_theme/theme/class-theme.php:
```diff
@@ -58,22 +36,22 @@ public function enqueue_styles() {
     wp_deregister_script( 'jquery-migrate' );
     wp_deregister_script( 'jquery' );
-    wp_register_script( 'jquery', get_template_directory_uri() . '/skin/public/scripts/vendors/jquery.min.js', array(), '3.3.1' );
+    wp_register_script( 'jquery', get_template_directory_uri() . '/skin/public/scripts/vendors/jquery.min.js', array(), '3.3.1' ); // phpcs:ignore WordPress.WP.EnqueuedResourceParameters.NotInFooter
```
Likewise about not being in the footer.

iruzevic replied:
in all WP themes, Jquery is never loaded in the footer so we shouldn't neither.

dingo-d replied:
Just because they do it, doesn't mean it7s the right way to do it 😄 They are still using bundled jQuery (1.2) xD

I think we should put our scripts in footer, and make them dependent on the bundled vendor jquery. This way we are not blocking down the HTML parsing on the page (which will happen if the script is in the header).

If we put them in the header, we should investigate in adding `async` and `defer` attributes on our scripts.

[Reference](https://developers.google.com/speed/docs/insights/BlockingJS)

iruzevic replied:
I agree with you 100%, but we can't guarantee that plugin developers will hook their scripts to the footer or dependent on jQuery, and if the script is loaded in header and jQuery is in the footer it won't work.

IvanGrginovInf replied:
How about adding it as an option? Radio button with "Load in header, load in header async, load in footer?"

dingo-d replied:
Not a bad idea, but I think that it would be better to leave that to the users, and just present the minimal working version of the developers to use and enhance :)

iruzevic commented:
- fixing merge conflicts with develop branch

dingo-d commented on wp-content/themes/inf_theme/admin/class-media.php:
```diff
@@ -77,22 +59,21 @@ public function enable_mime_types( $mimes ) {
+          $svg_content = implode( ' ', $svg_content );
 
-          if ( ! $this->general_helper->is_valid_xml( $svg_content ) ) {
+          if ( ! static::is_valid_xml( $svg_content ) ) {
```
Replace `static::` with `$this->` since it's not a static method.
